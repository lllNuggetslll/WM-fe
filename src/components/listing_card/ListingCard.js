import React from "react";
import { object, func } from "prop-types";
import Avatar from "../avatar";
import styled from "styled-components";
import get from "lodash.get";
import CardDetails from "./../card_details";
import { mobileBreakPoint } from "./../../constants/breakPoints";

const CardWrapper = styled.div`
  background: white;
  width: 100%;
  height: 66px;
  display: flex;
  align-items: center;
  margin-bottom: 10px;
  padding: 10px;
  width: 700px;
  padding: 20px;
  box-shadow: 1px 1px 5px -1px rgba(120, 120, 120, 1);
  border-radius: 3px 3px 0 0;
  cursor: pointer;

  &:hover {
    box-shadow: 1px 1px 5px 1px rgba(120, 120, 120, 1);
  }

  @media only screen and (max-width: ${mobileBreakPoint}) {
    width: 380px;
    flex-direction: row-reverse;
    justify-content: space-between;
    border-radius: 3px;
    align-items: initial;
    height: 100px;
  }
`;

const ListingCard = ({ listing, onClick }) => (
  <CardWrapper data-testid="CardWrapper" onClick={onClick}>
    <Avatar img={`${get(listing, "avatar_image.small_url")}`} />
    <CardDetails {...{ listing }} />
  </CardWrapper>
);

ListingCard.propTypes = {
  listing: object.isRequired,
  onClick: func.isRequired
};

export default ListingCard;
