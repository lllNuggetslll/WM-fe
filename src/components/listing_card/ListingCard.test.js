import React from "react";
import { shallow } from "enzyme";
import ListingCard from "./ListingCard";

describe("ListingCard", () => {
  const listing = {
    city: "city",
    distance: 1,
    name: "name",
    rating: 1
  };
  const onClick = jest.fn();

  it("should render", () => {
    const wrapper = shallow(<ListingCard {...{ listing, onClick }} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should fire onClick when clicked", () => {
    const wrapper = shallow(<ListingCard {...{ listing, onClick }} />);

    wrapper.find('[data-testid="CardWrapper"]').simulate("click");

    expect(onClick).toHaveBeenCalled();
  });
});
