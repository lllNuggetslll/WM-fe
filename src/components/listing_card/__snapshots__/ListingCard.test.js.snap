// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingCard should render 1`] = `
<styled.div
  data-testid="CardWrapper"
  onClick={[MockFunction]}
>
  <Avatar
    height="70px"
    img="undefined"
    width="70px"
  />
  <CardDetails
    listing={
      Object {
        "city": "city",
        "distance": 1,
        "name": "name",
        "rating": 1,
      }
    }
  />
</styled.div>
`;
